### PR TITLE
feat: correctly calculate done percentage

### PR DESCRIPTION
**Pull request by SgtPooki**

### CODE REVIEW

**Comment by SgtPooki** [comment]
pushed up just the test updates so we can see that the [original calculations](https://github.com/pln-planning-tools/StarMaps/blob/e72a4343c70d2468525144b02f3a792dd1a6326d/lib/backend/calculateCompletionRate.ts) fail with grandchildren. see https://github.com/pln-planning-tools/StarMaps/actions/runs/3570036790/jobs/6000617832

File: lib/backend/calculateCompletionRate.ts [hunk 1]
```diff
@@ -2,25 +2,53 @@ import { IssueStates } from '../enums';
...
+      return {
+        total,
+        open,
+        closed,
+      };
+    }, {open, closed, total});
+    open = withChildren.open;
+    closed = withChildren.closed;
+    total = withChildren.total;
+  }
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply lib/backend/calculateCompletionRate.ts:40]: we can improve it.. but I didn't want to focus on this level of improvement until I had async children done. the changes in this PR are a significant improvement over previous changes, and your changes are another improvement upon that. 

I think this would have been better handled as another issue, but I will fix and add a test for duplicate children.

**Review by SgtPooki — COMMENTED** [review]
self review
